electronics/circular_convolution.py

```diff
@@ -0,0 +1,99 @@
+# https://en.wikipedia.org/wiki/Circular_convolution
+
+"""
+Circular convolution, also known as cyclic convolution,
+is a special case of periodic convolution, which is the convolution of two
+periodic functions that have the same period. Periodic convolution arises,
+for example, in the context of the discrete-time Fourier transform (DTFT).
+In particular, the DTFT of the product of two discrete sequences is the periodic
+convolution of the DTFTs of the individual sequences. And each DTFT is a periodic
+summation of a continuous Fourier transform function.
+
+Source: https://en.wikipedia.org/wiki/Circular_convolution
+"""
+
+import doctest
+from collections import deque
+
+import numpy as np
+
+
+class CircularConvolution:
+    """
+    This class stores the first and second signal and performs the circular convolution
+    """
+
+    def __init__(self) -> None:
+        """
+        First signal and second signal are stored as 1-D array
+        """
+
+        self.first_signal = [2, 1, 2, -1]
+        self.second_signal = [1, 2, 3, 4]
+
+    def circular_convolution(self) -> list[float]:
+        """
+        This function performs the circular convolution of the first and second signal
+        using matrix method
+
+        Usage:
+        >>> import circular_convolution as cc
+        >>> convolution = cc.CircularConvolution()
+        >>> convolution.circular_convolution()
+        [10, 10, 6, 14]
+
+        >>> convolution.first_signal = [1, 0.5]
+        >>> convolution.second_signal = [0.5, 1]
+        >>> convolution.circular_convolution()
+        [1.0, 1.25]
+
+        >>> convolution.first_signal = [0.2, 0.4, 0.6, 0.8, 1.0, 1.2, 1.4, 1.6]
+        >>> convolution.second_signal = [0.1, 0.3, 0.5, 0.7, 0.9, 1.1, 1.3, 1.5]
+        >>> convolution.circular_convolution()
+        [5.2, 6.0, 6.48, 6.64, 6.48, 6.0, 5.2, 4.08]
+
```

Comment:
Add another unit test case where `first_signal` & `second_signal` have unequal sizes.

Reply:
Thanks for reviewing!
I will add the test case for unequal signals soon!